nix_update/version/github.py

```diff
@@ -1,38 +1,25 @@
+import json
 import re
 import urllib.request
 import xml.etree.ElementTree as ET
-from urllib.parse import ParseResult, unquote, urlparse
-from xml.etree.ElementTree import Element
+from urllib.parse import ParseResult, urlparse
 
-from ..errors import VersionError
 from ..utils import info
 from .version import Version
 
 
-def version_from_entry(entry: Element) -> Version:
-    if entry is None:
-        raise VersionError("No release found")
-    link = entry.find("{http://www.w3.org/2005/Atom}link")
-    assert link is not None
-    href = link.attrib["href"]
-    url = urlparse(href)
-    # TODO: set pre-release flag
-    return Version(unquote(url.path.split("/")[-1]))
-
-
 def fetch_github_versions(url: ParseResult) -> list[Version]:
     if url.netloc != "github.com":
         return []
     parts = url.path.split("/")
     owner, repo = parts[1], parts[2]
     repo = re.sub(r"\.git$", "", repo)
     # TODO fallback to tags?
-    feed_url = f"https://github.com/{owner}/{repo}/releases.atom"
-    info(f"fetch {feed_url}")
-    resp = urllib.request.urlopen(feed_url)
-    tree = ET.fromstring(resp.read())
-    releases = tree.findall(".//{http://www.w3.org/2005/Atom}entry")
-    return [version_from_entry(x) for x in releases]
+    github_url = f"https://api.github.com/repos/{owner}/{repo}/releases"
```

Comment:
When using the github api, we would also need to support using a github token i.e. through environment variables. Otherwise it might fail to fetch updates i.e. when users are behind a shared ip or in CI like github because of rate limits.

I would suggest to keep both the RSS feed and also use the API if a GITHUB_TOKEN is provided.

Reply:
@jvanbruegge what do you think?

Reply:
Yes, I can do it. Just theblast few weeks were quite stressfull and I will probably not get to it before the end of next week